catalog/app/utils/bucketPreferences.ts

```diff
@@ -0,0 +1,109 @@
+import * as R from 'ramda'
+
+import * as AWS from 'utils/AWS'
+import { useData } from 'utils/Data'
+import { makeSchemaValidator } from 'utils/json-schema'
+import yaml from 'utils/yaml'
+import bucketPreferencesSchema from 'schemas/bucketConfig.yml.json'
+import * as bucketErrors from 'containers/Bucket/errors'
+import * as requests from 'containers/Bucket/requests'
+
+export type ActionPreferences = Record<
+  'copyPackage' | 'createPackage' | 'revisePackage',
+  boolean
+>
+
+export type NavPreferences = Record<'files' | 'packages' | 'queries', boolean>
+
+type UiPreferences = {
+  actions: ActionPreferences
+  nav: NavPreferences
+}
+
+interface BucketPreferencesYaml {
+  ui?: Partial<UiPreferences>
+}
+
+export interface BucketPreferences {
+  ui: UiPreferences
+}
+
+export const defaultPreferences: BucketPreferences = {
+  ui: {
+    actions: {
+      copyPackage: true,
+      createPackage: true,
+      revisePackage: true,
+    },
+    nav: {
+      files: true,
+      packages: true,
+      queries: true,
+    },
+  },
+}
+
+const bucketPreferencesValidator = makeSchemaValidator(bucketPreferencesSchema)
+
+function validate(data: unknown): asserts data is BucketPreferencesYaml {
+  const errors = bucketPreferencesValidator(data)
+  if (errors.length) throw new bucketErrors.BucketPreferencesInvalid({ errors })
+}
+
+export function parse(bucketPreferencesYaml: string): BucketPreferences {
+  const data = yaml(bucketPreferencesYaml)
+  if (!data) return defaultPreferences
+
+  validate(data)
+
+  return {
+    ui: {
+      actions: R.mergeRight(defaultPreferences.ui.actions, data?.ui?.actions || {}),
+      nav: R.mergeRight(defaultPreferences.ui.nav, data?.ui?.nav || {}),
+    },
+  }
+}
+
+const BUCKET_PREFERENCES_PATH = '.quilt/catalog/config.yml'
```

Comment:
it's still not 100% clear which extension should be used: we already use `yml` for workflows, but `yaml` is a recommended option

Reply:
It's recommended, just like /dʒɪf/ is recommended spelling of "gif" :)
`.yml` is widely used, and I think it's more predictable.

Reply:
I use both. And unconsciously tend to use `.yaml` at formal documents and `.yml` in the wild.
I separate this discussion to Slack